### PR TITLE
[ADD] pylint_odoo: Adding check detected <data> tag inside <odoo> tag and show warning 

### PR DESCRIPTION
**Pull request by JesusZapata**
The new check detected <data> tag inside <odoo> tag and show warning

Related with https://github.com/Vauxoo/pylint-odoo/pull/117

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +34,7 @@ install:
     - ${TRAVIS_BUILD_DIR}/install.sh
 
     # Install testing dependencies
+    - pip install six
```
Comment (pedrobaeza) [inline .travis.yml]: Why we need this?

Comment (JesusZapata) [inline .travis.yml]: @pedrobaeza 
Sorry, I deleted this not related change
Thanks

**Comment by sbidoul** [comment]
Is there a written Odoo convention somewhere about this?

I'd think it is legit to have several data sections in the same file, some with noupdate=0 and others with noupdate=1.

**Comment by moylop260** [comment]
Hello @sbidoul
Yes, we can.

Good cases:
=========

```xml
<odoo noupdate="0">
</odoo>

<odoo noupdate="1">
</odoo>

<odoo>
</odoo>
```

Bad cases:
========
```xml
<odoo>
    <data noupdate="0">
    </data>
</odoo>

<odoo>
    <data noupdate="1">
    </data>
</odoo>

<odoo>
    <data>
    </data>
</odoo>
```

Examples from odoo:
  - [10.0 - addons/crm/data/crm_stage_data.xml#L2](https://github.com/odoo/odoo/blame/65e49baedeeab52ad9ba9c1e001449527118197a/addons/crm/data/crm_stage_data.xml#L2)
 - [10.0 - addons/payment/data/payment_acquirer_data.xml#L2](https://github.com/odoo/odoo/blame/65e49baedeeab52ad9ba9c1e001449527118197a/addons/payment/data/payment_acquirer_data.xml#L2)
 - [10.0 - addons/website_sale/data/demo.xml#L2](https://github.com/odoo/odoo/blame/65e49baedeeab52ad9ba9c1e001449527118197a/addons/website_sale/data/demo.xml#L2)

Use `<odoo noupdate="1">` node instead of `<odoo><data noupdate="1">`

Currently we have a convention to use 4 spaces for xml files, using the additional node `noupdate` force us to remove or add indentation.

Example:

Diffs using just `odoo` node (don't require add a new node and don't require re-ident):

```diff
-<odoo>
+<odoo noupdate="1">
    <record 1...
    <record 2...
    <record 3...
</odoo>
```

Diff using `odoo/data` node (re-ident is required if data wasn't defined to add `noupdate=1`):

```diff
<odoo>
+    <data noupdate="1">
-    <record 1...
+        <record 1...
-     <record 2...
+        <record 2...
-    <record 3...
+        <record 3...
+    </data>
</odoo>
```

**Comment by sbidoul** [comment]
Your Good cases example is not valid XML document: one cannot have several root elements in an XML file. I don't know if Odoo accepts that but even if it does it is definitely bad practice.

**Comment by sbidoul** [comment]
@moylop260 sorry, I misread your example. What I mean is somthing like this should be accepted:

```xml
<odoo>
  <data noupdate="0">
  </data>
  <data noupdate="1">
  </data>
</odoo>
```

**Comment by JesusZapata** [comment]
@sbidoul 
I understand the point.
I made the changes for support your case.
Could do you approve this change?

**Comment by sbidoul** [comment]
@JesusZapata thanks!

BTW, are you planning to do a new release pylint-odoo to pypi?

**Comment by sbidoul** [comment]
> on MQT, as it takes latest git version.

@pedrobaeza I know I know. The world is bigger than MQT, though :wink: And from time to time a release that pip can cache is nice to have.

**Comment by pedrobaeza** [comment]
Yes, of course. We can indeed publish a new release for "the rest of the world", hehe.

**Comment by dreispt** [comment]
LGTM, but please don't forget to add the new check to the README doc.

**Comment by JesusZapata** [comment]
@dreispt 
Done, Thanks for you comment

File: pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+    </data>
+    <data>
+    </data>
+</odoo>
```
Comment (dreispt) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: Could you please add a test case with the [example](https://github.com/OCA/pylint-odoo/pull/129#issuecomment-306834319) provided by Stéphane?

```xml
<odoo>
  <data noupdate="0">
  </data>
  <data noupdate="1">
  </data>
</odoo>
```

Comment (JesusZapata) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: @dreispt 
The case when `<data noupdate="0">` is equal to only the tag `<data>`
The default valud of `noupdate` is `0`
Therefore I add only the `<data>` tag

Are you agree?

Comment (dreispt) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: Why is it called "broken_module2"? It is a correct example.

Comment (dreispt) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: AFAICS this case [does not pass pylint checks](https://travis-ci.org/OCA/pylint-odoo/jobs/240797733#L809) but it should.

Comment (JesusZapata) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: @dreispt 
Sorry, I don't understand
The file [broken_module2/tests/data/help_test_data.xml](https://github.com/OCA/pylint-odoo/blob/master/pylint_odoo/test_repo/broken_module2/tests/data/help_test_data.xml) has a `<data>` tag (only one) inside another tag `<odoo>`. That is the purpose of this check

The files with the examples were saved in `broken_module2` and `broken_module` only to use the same existing folders.

There is no special reason

Comment (dreispt) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: Sorry, I missed a link. [Stéphane's case](https://github.com/OCA/pylint-odoo/pull/129#issuecomment-306834319) should pass checks and not raise a message, right?:

```xml
<odoo>
  <data noupdate="0">
  </data>
  <data noupdate="1">
  </data>
</odoo>
```

Comment (JesusZapata) [inline pylint_odoo/test_repo/broken_module2/tests/data/odoo_data_noupdate_1.xml]: @dreispt 
Yes, the message only show when inside the `<odoo>` tag contains **_only one_** `<data>` tag

The detected cases are

```xml
<odoo>
  <data noupdate="0">
    ...
  </data>
</odoo>
```
```xml
<odoo>
  <data noupdate="1">
    ...
  </data>
</odoo>
```
```xml
<odoo>
  <data>
    ...
  </data>
</odoo>
```

**Pull request merged**